chain.go

```diff
@@ -154,3 +204,121 @@ func RevealChain(c *Chain) (string, error) {
 	}
 	return r.Entry, nil
 }
+
+// GetChainHead returns the hash of the most recent Entry made into a given
+// Factom Chain.
+func GetChainHead(chainid string) (string, bool, error) {
+	params := chainIDRequest{ChainID: chainid}
+	req := NewJSON2Request("chain-head", APICounter(), params)
+	resp, err := factomdRequest(req)
+	if err != nil {
+		return "", false, err
+	}
+	if resp.Error != nil {
+		return "", false, resp.Error
+	}
+
+	head := new(struct {
+		ChainHead          string `json:"chainhead"`
+		ChainInProcessList bool   `json:"chaininprocesslist"`
+	})
+	if err := json.Unmarshal(resp.JSONResult(), head); err != nil {
+		return "", false, err
+	}
+
+	return head.ChainHead, head.ChainInProcessList, nil
+}
+
+// GetAllChainEntries returns a list of all Factom Entries for a given Chain.
+func GetAllChainEntries(chainid string) ([]*Entry, error) {
+	es := make([]*Entry, 0)
+
+	head, inPL, err := GetChainHead(chainid)
+	if err != nil {
+		return es, err
+	}
+
+	if head == "" && inPL {
+		return nil, ErrChainPending
+	}
+
+	for ebhash := head; ebhash != ZeroHash; {
+		eb, err := GetEBlock(ebhash)
+		if err != nil {
+			return es, err
+		}
+		s, err := GetAllEBlockEntries(ebhash)
+		if err != nil {
+			return es, err
+		}
+		es = append(s, es...)
+
+		ebhash = eb.Header.PrevKeyMR
+	}
+
+	return es, nil
+}
+
+// GetAllChainEntriesAtHeight returns a list of all Factom Entries for a given
+// Chain at a given point in the Chain's history.
+func GetAllChainEntriesAtHeight(chainid string, height int64) ([]*Entry, error) {
+	es := make([]*Entry, 0)
```

Comment:
I feel this could be refactored to share most of the code with `GetAllChainEntries`